internal/internal_activity.go

```diff
@@ -90,14 +91,19 @@ type (
 	}
 
 	activityEnvironment struct {
-		taskToken         []byte
-		workflowExecution WorkflowExecution
-		activityID        string
-		activityType      ActivityType
-		serviceInvoker    ServiceInvoker
-		logger            *zap.Logger
-		metricsScope      tally.Scope
-		isLocalActivity   bool
+		taskToken          []byte
+		workflowExecution  WorkflowExecution
+		activityID         string
+		activityType       ActivityType
+		serviceInvoker     ServiceInvoker
+		logger             *zap.Logger
+		metricsScope       tally.Scope
+		isLocalActivity    bool
+		heartbeatTimeout   time.Duration
```

Comment:
How about embedding activityInfo in here instead of duplicating the fields?

Reply:
I think it is mostly about layering. ActivityInfo doesn't exist at the level of this function.

Reply:
Not sure what you mean? They're both in the same package, struct definitions can be embedded.